doc/rtd/development/datasource_creation.rst

```diff
@@ -0,0 +1,89 @@
+.. _datasource_creation:
+
+Datasource creation
+*******************
+
+Since there are multiple ways to provide `user data`, `meta data`, and
+`vendor data` (each cloud solution seems to prefer its own way), a datasource
+abstract class was internally created to allow for a single way to access the
+different cloud systems methods, providing this data through the typical usage
+of subclasses. See :file:`cloud-init/sources/__init__.py` as a starting
```

Comment:
That sentence is a bunch of fluff and jargon.

How about:
```
An abstract base class defines a single interface to interact with the different clouds. Each cloud implementation must inherit from this base class to utilize this shared functionality and interface.```

Reply:
Definitely better. I'd say "use" instead of "utilize", and I'd keep "datasource" with the "abstract base class" bit (so that we're being specific about which abstract base class we're talking about), but otherwise, yeah, much improved.